pennylane/qchem/vibrational/bosonic_mapping.py

```diff
@@ -0,0 +1,160 @@
+
+from .bosonic import BoseSentence, BoseWord
+
+
+def _get_pauli_op(i, j, qub_id):
+    r"""Returns expression to convert qubit-local term ::math::``\ket{x_j}\bra{x^{'}}``
+    to qubit operators as given in Eq. (6-9) <https://www.nature.com/articles/s41534-020-0278-0>"""
+
+    c1, c2 = 0.5, -0.5 if i == 1 else 0.5
+
+    if i != j:
+        return PauliSentence({PauliWord({qub_id: "X"}): c1, PauliWord({qub_id: "Y"}): c2 * 1j})
+
+    return PauliSentence({PauliWord({}): c1, PauliWord({qub_id: "Z"}): c2})
+
+
+def binary_mapping(
+    bose_operator: Union[BoseWord, BoseSentence],
+    nstates: int = 2,
+    ps: bool = False,
+    wire_map: dict = None,
+    tol: float = None,
+):
+    r"""Convert a bosonic operator to a qubit operator using the standard-binary mapping.
+
+    The mapping procedure is described in `arXiv:1507.03271 <https://arxiv.org/pdf/1507.03271>`_.
+
+    Args:
+        bose_operator(BoseWord, BoseSentence): the bosonic operator
+        nstates(int): maximum number of states a boson can occupy
+        ps (bool): whether to return the result as a PauliSentence instead of an
+            operator. Defaults to False.
+        wire_map (dict): a dictionary defining how to map the states of
+            the Bose operator to qubit wires. If None, integers used to
+            label the bosonic states will be used as wire labels. Defaults to None.
+        tol (float): tolerance for discarding the imaginary part of the coefficients
+
+    Returns:
+        a linear combination of qubit operators
+
+    **Example**
+
+    >>> w = qml.qchem.BoseWord({(0, 0): "+"})
+    >>> qml.qchem.binary_mapping(w, nstates=4)
+    0.6830127018922193 * X(0)
+    + -0.1830127018922193 * X(0) @ Z(1)
+    + -0.6830127018922193j * Y(0)
+    + 0.1830127018922193j * Y(0) @ Z(1)
+    + 0.3535533905932738 * X(0) @ X(1)
+    + -0.3535533905932738j * X(0) @ Y(1)
+    + 0.3535533905932738j * Y(0) @ X(1)
+    + (0.3535533905932738+0j) * Y(0) @ Y(1)
+    """
+
+    qubit_operator = _binary_mapping_dispatch(bose_operator, nstates, tol=tol)
+
+    wires = list(bose_operator.wires) or [0]
+    identity_wire = wires[0]
+    if not ps:
+        qubit_operator = qubit_operator.operation(wire_order=[identity_wire])
+
+    if wire_map:
+        return qubit_operator.map_wires(wire_map)
+
+    return qubit_operator
+
+
+@singledispatch
+def _binary_mapping_dispatch(bose_operator, nstates, tol):
+    """Dispatches to appropriate function if bose_operator is a BoseWord or BoseSentence."""
+    raise ValueError(f"bose_operator must be a BoseWord or BoseSentence, got: {bose_operator}")
+
+
+@_binary_mapping_dispatch.register
+def _(bose_operator: BoseWord, nstates, tol=None):
+
+    if nstates < 2:
+        raise ValueError(
+            f"Number of states a boson can occupy cannot be less than 2, provided {nstates}."
+        )
+    nqub_per_boson = int(np.ceil(np.log2(nstates)))
+
+    cr_op = np.zeros((nstates, nstates))
```

Comment:
Could you just use creation instead of cr_op, if it means the creation operator? Or any other descriptive name.